packages/@expo/metro-config/src/babel-transformer.ts

```diff
@@ -114,6 +115,10 @@ function getBabelCaller({
 
     isHMREnabled: options.hot,
 
+    // Pass on the input type. Scripts shall be transformed to avoid dependencies (imports/requires),
+    // for example by polyfills or Babel runtime
+    type: options.type,
```

Comment:
`metroModuleType` maybe

Reply:
+1 for this. the `getType` in babel-preset-expo isn't clear